### PR TITLE
Added StringHelper::ensurePrefix

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | none

### CODE REVIEW

File: framework/helpers/BaseStringHelper.php [hunk 1]
```diff
@@ -350,4 +350,25 @@ public static function base64UrlDecode($input)
...
+     * @param string $prefix Prefix to be ensured in the string
+     * @param bool $caseSensitive Case sensitive comparison. Default is true. When case sensitive is enabled, $prefix
+     * must exactly match the starting of the string.
+     * @return string
+     *
+     * @since 2.0.13
+     */
+    public static function ensurePrefix($string, $prefix, $caseSensitive = true)
+    {
+        if (!self::startsWith($string, $prefix, $caseSensitive)) {
```
Comment (rob006) [inline framework/helpers/BaseStringHelper.php]: `self` -> `static` ?

Comment (samdark) [reply framework/helpers/BaseStringHelper.php]: Yes.

**Review by dmirogin — APPROVED** [review]

File: framework/helpers/BaseStringHelper.php [hunk 2]
```diff
@@ -350,4 +350,25 @@ public static function base64UrlDecode($input)
     {
         return base64_decode(strtr($input, '-_', '+/'));
     }
+
+    /**
+     * Ensures string is prefixed with the prefix specified. Avoids double prefixing.
+     *
+     * Useful for adding "http://" to user entered website addresses, twitter handles etc.
```
Comment (cebe) [inline framework/helpers/BaseStringHelper.php]: how about http vs. https?

File: framework/helpers/BaseStringHelper.php [hunk 3]
```diff
@@ -350,4 +350,25 @@ public static function base64UrlDecode($input)
...
+     * @param bool $caseSensitive Case sensitive comparison. Default is true. When case sensitive is enabled, $prefix
+     * must exactly match the starting of the string.
+     * @return string
+     *
+     * @since 2.0.13
+     */
+    public static function ensurePrefix($string, $prefix, $caseSensitive = true)
+    {
+        if (!static::startsWith($string, $prefix, $caseSensitive)) {
+            $string = $prefix . $string;
```
Comment (cebe) [inline framework/helpers/BaseStringHelper.php]: no need to modify the string, you can just return the result.

File: framework/helpers/BaseStringHelper.php (hunk 2 above)
Comment (cebe) [reply framework/helpers/BaseStringHelper.php]: so its not really useful for user content, what are other use cases?

**Review by cebe — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
Closing for the reason @cebe described. Proper handling is more complex than that...

**Pull request closed without merging**